### PR TITLE
Fetching articles from sections

### PR DESCRIPTION
**Pull request by filipgolonka**
@Schibsted-Tech-Polska/sa

### CODE REVIEW

File: spec/Snt/Capi/Repository/Article/FindBySectionParametersSpec.php [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    const PUBLICATION_ID = 'sa';
+
+    const SECTION_NAME = 'bolig';
+
+    function let()
+    {
+        $this->beConstructedThrough('createForPublicationIdAndSections', [self::PUBLICATION_ID, [self::SECTION_NAME]]);
+    }
+
+    function it_can_be_constructed_through_named_constructor()
```
Comment (rafalkanski) [inline spec/Snt/Capi/Repository/Article/FindBySectionParametersSpec.php]: `it_is_initializable`?

File: spec/Snt/Capi/Repository/Article/FindBySectionParametersSpec.php [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    {
+        $this->shouldHaveType(FindBySectionParameters::class);
+        $this->shouldHaveType(FindParametersInterface::class);
+    }
+
+    function it_builds_url()
+    {
+        $path = sprintf(self::URL_PATTERN, self::PUBLICATION_ID, self::SECTION_NAME, '');
```
Comment (rafalkanski) [inline spec/Snt/Capi/Repository/Article/FindBySectionParametersSpec.php]: to be honest, i would create constant for last parameter and use it as `self::EMPTY_QUERY` (or similar). with that i don't need to scroll up and see how `URL_PATTERN` looks like , to understand what `''` means. what do you think?

**Comment by barkahl** [comment]
👍

File: spec/Snt/Capi/Repository/Article/FindBySectionParametersSpec.php (hunk 2 above)
Comment (filipgolonka) [inline spec/Snt/Capi/Repository/Article/FindBySectionParametersSpec.php]: :+1:

File: src/Snt/Capi/Repository/Article/FindBySectionParameters.php [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+    /**
+     * @param int $hotnessTo
+     *
+     * @return $this
+     * @throw InvalidArgumentException
+     */
+    public function setHotnessTo($hotnessTo)
+    {
+        $hotnessTo = (int) $hotnessTo;
+        if ($hotnessTo < 0 || $hotnessTo > 100) {
```
Comment (rafalkanski) [inline src/Snt/Capi/Repository/Article/FindBySectionParameters.php]: four times same condition - maybe private method `isNotBetweenZeroAndOneHundred($value)` or `valueIsNotBetween($value, $from, $to)` what do you think?

**Comment by rafalkanski** [comment]
👍  👏  👏  👏

File: src/Snt/Capi/Repository/Article/FindBySectionParameters.php [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+        }
+
+        return $this;
+    }
+
+    /**
+     * @param int $hotnessFrom
+     *
+     * @return $this
+     * @throw InvalidArgumentException
```
Comment (arkadiuszmalina) [inline src/Snt/Capi/Repository/Article/FindBySectionParameters.php]: should be `throws`

File: src/Snt/Capi/Repository/Article/FindBySectionParameters.php [hunk 5]
```diff
@@ -0,0 +1,227 @@
...
+        $this->parameters['hotnessFrom'] = $hotnessFrom;
+
+        return $this;
+    }
+
+    /**
+     * @param int $hotnessTo
+     *
+     * @return $this
+     * @throw InvalidArgumentException
```
Comment (arkadiuszmalina) [inline src/Snt/Capi/Repository/Article/FindBySectionParameters.php]: as above

File: src/Snt/Capi/Repository/Article/FindBySectionParameters.php [hunk 6]
```diff
@@ -0,0 +1,227 @@
...
+        $this->parameters['hotnessTo'] = $hotnessTo;
+
+        return $this;
+    }
+
+    /**
+     * @param int $lifetimeFrom
+     *
+     * @return $this
+     * @throw InvalidArgumentException
```
Comment (arkadiuszmalina) [inline src/Snt/Capi/Repository/Article/FindBySectionParameters.php]: as above

File: src/Snt/Capi/Repository/Article/FindBySectionParameters.php [hunk 7]
```diff
@@ -0,0 +1,227 @@
...
+        $this->parameters['lifetimeFrom'] = $lifetimeFrom;
+
+        return $this;
+    }
+
+    /**
+     * @param int $lifetimeTo
+     *
+     * @return $this
+     * @throw InvalidArgumentException
```
Comment (arkadiuszmalina) [inline src/Snt/Capi/Repository/Article/FindBySectionParameters.php]: as above

**Comment by arkadiuszmalina** [comment]
👍  except above

**Comment by wsromek** [comment]
👍

**Pull request merged**